### ISSUE TITLE
How do I tell ipaas to trust some SSL certificate?

### ISSUE THREAD

**Issue by jludvice** [syndesisio/syndesis-ui#570]
What do I do if I need to call http POST to an endpoint that uses SSL and it's certificate is signed by corporate CA (which is not trusted by Java by default) ?
Or what if it's a self-signed certificate?

How do I connect to such endpoint?

**Comment by jludvice** [comment]
Java developers might have experience with preparing trustStore and similar stuff but how do we get this done in user friendly way?

Would it make sense to autodetect if given URL starts with HTTPS and 
* build truststore on behalf of the user
* show him the PKI chain and ask him if he really trusts those certificates

**Comment by gashcrumb** [comment]
Yeah, from javascript you can't do any of this, the browser just blocks the request.  The only way is by actually connecting to the endpoint, i.e. change the location to navigate to a page at that endpoint, which will then allow the browser to prompt the user that it's not a secure location but gives the user a chance to inspect the certificate and add an exception.  So then such an endpoint could probably then just redirect back to the original URL.

The only other option is to use a certificate signed by a known CA, or import the CA of that certificate into the browser's trusted CA list.

**Comment by jimmidyson** [comment]
I'd expect this to be server-side, not from browser.

**Comment by gashcrumb** [comment]
Ah gotcha, so posting via the http4 component to some endpoint, thanks for clarifying.

**Comment by jludvice** [comment]
My inspiration was [KeyStore Explorer](http://keystore-explorer.org/) (GNU/GPL license).
It has a way to Examine > Examine SSL 

I just paste URL and port and it shows me the certificate chain and allows me to import those certificates into keystore.
![screenshot_20170606_153905](https://cloud.githubusercontent.com/assets/8707241/26832051/77171f78-4ace-11e7-95a1-4ebbda7d06bb.png)

Could we do this automagically server-side?

**Comment by jludvice** [comment]
One alternatives for trustStore comes to my mind:
* copy required certificates into `/etc/pki/ca-trust/source/anchors` 
* and run `update-ca-trust` and 

They should end up in default JVM trust store that way.

This worked for me when I needed wildfly to trust LDAP server.

**Comment by jimmidyson** [comment]
@jludvice That's one idea but I think we can do it easier via an init container to download the certificate to a shared volume and use my pemtokeystore docker image to create a truststore on the fly. We do something similar for OpenShift's service serving certificates.

I guess the missing piece is where we slot this in the UI flow.

**Comment by gashcrumb** [comment]
Are there any tasks to do on this issue or can it be closed?

**Comment by jludvice** [comment]
@gashcrumb I wanted to know how can user create http connection to some self signed rest api.
The http connector is broken anyway, so this issue is out of scope imho.

**Comment by gashcrumb** [comment]
Gotcha, yeah, from the sound of it this would really need an overarching scenario as we'll probably need to add features to the connector and expose them in the UI.  So this should probably move to syndesis-project as a user story.

**Comment by zregvart** [comment]
Yeah, this would be a connector feature that we expose in the UI via properties. Camel has a number of properties in common to HTTP* components that govern TLS configuration that we can expose from the connector.
